erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -604,6 +635,9 @@ def update_component_row(self, struct_row, amount, key, overwrite=1, additional_
 			component_row.amount = amount
 			component_row.deduct_full_tax_on_selected_payroll_date = struct_row.deduct_full_tax_on_selected_payroll_date
 
+	def get_overwritten_components(self, key):
+		return [ele.salary_component for ele in self.get(key) if ele.overwritten == 1]
```

Comment:
```suggestion
		return [entry.salary_component for entry in self.get(key) if entry.overwritten]
```
Use `entry` instead of `ele`. No need to explicitly check whether `ele.overwritten == 1`. `entry.overwritten` will evaluate to True.